### PR TITLE
Discord login

### PR DESCRIPTION
**Pull request by bates64**

### CODE REVIEW

File: src/hooks.server.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    return resolve(event);
+}
+
+export const handle: Handle = sequence(
+    SvelteKitAuth({
+        providers: [
+            Discord({
+                clientId: "1113277158543073381",
```
Comment (bates64) [inline src/hooks.server.ts]: Should be from env

File: src/lib/components/Signin.svelte [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    let showSignOutButton = false;
+
+    function signIn() {
+        window.location.href = "/auth/signin";
+    }
+
+    function signOut() {
+        window.location.href = "/auth/signout";
+    }
```
Comment (bates64) [inline src/lib/components/Signin.svelte]: There may be a "proper" way to do redirects with sveltekit, can you check?

File: src/lib/components/Signin.svelte [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+</script>
+
+{#if $page.data.session}
+    {#if $page.data.session.user?.image}
+        <div
+            class="user-icon"
+            style="background-image: url('{$page.data.session.user.image}');"
```
Comment (bates64) [inline src/lib/components/Signin.svelte]: I think `style:background-image=` is better

File: src/lib/components/Signin.svelte [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            on:click={() => (showSignOutButton = true)}
```
Comment (bates64) [inline src/lib/components/Signin.svelte]:
```suggestion
            on:click={() => {showSignOutButton = true}}
```

File: src/lib/components/Signin.svelte [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        background-color: #444;
+        color: #fff;
+        padding: 10px;
+        border-radius: 5px;
+    }
+    .sign-out-button {
+        position: fixed;
+        top: 80px;
+        right: 20px;
+    }
```
Comment (bates64) [inline src/lib/components/Signin.svelte]: Look into doing this with https://tailwindcss.com/ instead please

File: src/routes/+page.svelte [hunk 6]
```diff
@@ -3,7 +3,7 @@
 
     import LandingHeader from "./LandingHeader.svelte";
     import LinkButton from "./LinkButton.svelte";
-
+    import Signin from "$lib/components/Signin.svelte";
```
Comment (bates64) [inline src/routes/+page.svelte]: Can you name the component `SignIn` (i.e. capitalise "in")

File: src/lib/components/Signin.svelte (hunk 2 above)
Comment (bates64) [reply src/lib/components/Signin.svelte]: goto from $app/navigation

Comment (bates64) [reply src/lib/components/Signin.svelte]: Unless this breaks auth.js

File: src/routes/mods/+page.svelte [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+	link: "#",
+	},
+	];
+
+	function signIn() {
+	window.location.href = "/auth/signin";
+	}
+	function signOut() {
+	window.location.href = "/auth/signout";
+	}
```
Comment (bates64) [inline src/routes/mods/+page.svelte]: Needs formatting, do `pnpm run format` and it should hopefully fix this
if not thats a bug 💀

File: src/routes/mods/+page.svelte [hunk 8]
```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+	import Breadcrumbs from "$lib/components/Breadcrumbs.svelte";
+	import { page } from "$app/stores";
+	let showSignInButton = false;
+	import Signin from "$lib/components/Signin.svelte";
```
Comment (bates64) [inline src/routes/mods/+page.svelte]: put imports at the top like
```suggestion
	import Signin from "$lib/components/Signin.svelte";
    let showSignInButton = false;
```

File: src/routes/mods/+page.svelte [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+    .mod-name a {
+        text-decoration: none;
+        color: inherit;
+    }
+
+    .mod-description {
+        color: #666;
+    }
+</style>
```
Comment (bates64) [inline src/routes/mods/+page.svelte:80]: again tailwind where possible

File: src/lib/components/Signin.svelte (hunk 2 above)
Comment (L5050) [reply src/lib/components/Signin.svelte]: theres the goto function but it sometimes doesn't redirect properly

File: src/routes/+page.svelte (hunk 6 above)
Comment (L5050) [reply src/routes/+page.svelte]: Sure yeah

**Review by L5050 — APPROVED** [review]
adds new login features and a profile page

**Pull request merged**